lms/djangoapps/courseware/views.py

```diff
@@ -892,7 +895,18 @@ def course_about(request, course_id):
                     shoppingcart.models.CourseRegCodeItem.contained_in_order(cart, course_key)
 
             reg_then_add_to_cart_link = "{reg_url}?course_id={course_id}&enrollment_action=add_to_cart".format(
-                reg_url=reverse('register_user'), course_id=urllib.quote(str(course_id)))
+                reg_url=reverse('register_user'), course_id=urllib.quote(str(course_id))
+            )
+
+        ecommerce_checkout_link = ''
```

Comment:
Here I'm assigning values to ``mode`` and ``ecommerce_checkout_link`` because otherwise they are only assigned in the below if-block and referencing them in the ``render_to_response`` call would result in an error if they're not assigned.

The ``course_about`` view shows the about page of a course and button to enroll, or add the course to the shopping cart if it's a (no_id)professional one. If the ecommerce checkout service is enabled (and it's a (no_id)professional course) new links are constructed for that course and SKU that point the 'add to cart' button to Otto instead of the old shopping cart.